sdks/unity/AgonesSdk.cs

```diff
@@ -360,17 +360,37 @@ private void OnRequestCompleted(AsyncOperation _)
         private class GameServerHandler : DownloadHandlerScript
         {
             private WatchGameServerCallback callback;
+            private StringBuilder stringBuilder;
+
             public GameServerHandler(WatchGameServerCallback callback)
             {
                 this.callback = callback;
+                this.stringBuilder = new StringBuilder();
             }
 
             protected override bool ReceiveData(byte[] data, int dataLength)
             {
-                string json = Encoding.UTF8.GetString(data);
-                var dictionary = (Dictionary<string, object>) Json.Deserialize(json);
-                var gameServer = new GameServer(dictionary["result"] as Dictionary<string, object>);
-                this.callback(gameServer);
+                string dataString = Encoding.UTF8.GetString(data);
+                this.stringBuilder.Append(dataString);
+
+                string bufferString = stringBuilder.ToString();
+                int newlineIndex;
+
+                while ((newlineIndex = bufferString.IndexOf('\n')) >= 0)
+                {
+                    string fullLine = bufferString.Substring(0, newlineIndex);
+                    try
+                    {
+                        var dictionary = (Dictionary<string, object>) Json.Deserialize(fullLine);

```

Comment:
What does the input look like for this method? Is all the data expected to be on one line?

Reply:
fullLine should always be a single JSON message on a single line. grpc-gateway turns streams into newline delineated JSON messages and JSON encoding will escape any newlines in the data, so the only newlines we encounter will be the ones intentionally sent to break full JSON messages.